sys/signal.c

```diff
@@ -0,0 +1,158 @@
+#define KL_LOG KL_SIGNAL
+#include <klog.h>
+#include <signal.h>
+#include <thread.h>
+#include <malloc.h>
+#include <stdc.h>
+#include <errno.h>
+#include <sysent.h>
+#include <proc.h>
+
+static sighandler_t *def_sigact[NSIG] = {
```

Comment:
I reckon you will never store real pointers to signal handlers in this array. That's why they solved it in [slightly different way](http://bxr.su/FreeBSD/sys/kern/kern_sig.c#183), which I find less confusing.